elide-graphql/src/main/java/com/yahoo/elide/graphql/Environment.java

```diff
@@ -57,9 +57,7 @@ public Environment(DataFetchingEnvironment environment) {
 
         if (isRoot()) {
             // Flush (but don't commit) between root queries
-            requestScope.getDirtyResources().forEach(
-                    resource -> requestScope.getTransaction().save(resource.getObject(), requestScope)
-            );
+            requestScope.saveOrCreateObjects();
```

Comment:
This seems sane to me. Do you recall why we were deferring saves here?

Reply:
Saves are deferred to prevent cascades with uninitialized objects.  Elide first initializes everything in memory - and then triggers persist/merge - which then causes cascades.  That's my understanding.